tests/async.test.tsx

```diff
@@ -625,3 +625,70 @@ it('Handles synchronously invoked async set (#375)', async () => {
   await findByText('loading')
   await findByText('great document')
 })
+
+it('async write self atom', async () => {
```

Comment:
I think we are missing two test:

- A sync write self atom (in basic.test.tsx ofc)
- Error handling for self write - What's the expected value of the atom after you have handled the error boundary?